modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/fork/HandlerSuite.scala

```diff
@@ -0,0 +1,51 @@
+package org.tessellation.sdk.infrastructure.fork
+
+import cats.effect.IO
+
+import org.tessellation.schema.generation.Generation
+import org.tessellation.schema.generators.peerIdGen
+import org.tessellation.schema.gossip._
+import org.tessellation.sdk.config.types.ForkInfoStorageConfig
+import org.tessellation.sdk.domain.fork.{ForkInfo, StoredForkInfo}
+import org.tessellation.sdk.infrastructure.fork.ForkInfoStorageSuite.genStoredForkInfoEntry
+import org.tessellation.sdk.infrastructure.fork.handler.forkInfoHandler
+
+import eu.timepit.refined.auto._
+import io.circe.syntax.EncoderOps
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object HandlerSuite extends SimpleIOSuite with Checkers {
+
+  test("recieved gossip gets added to the storage") {
+    val gen = for {
+      forkInfo <- genStoredForkInfoEntry.map(_.forkInfo)
+      peerId1 <- peerIdGen
+      peerId2 <- peerIdGen
+    } yield (forkInfo, peerId1, peerId2)
+
+    forall(gen) {
+      case (forkInfo, peerId1, peerId2) =>
+        for {
+          storage <- ForkInfoStorage.make(ForkInfoStorageConfig(10))
+          handler = forkInfoHandler(storage)
+          rawRumor = PeerRumorRaw(
+            peerId1,
+            Ordinal(Generation.MinValue, Counter.MinValue),
+            forkInfo.asJson,
+            ContentType.of[ForkInfo]
+          )
+
+          _ <- handler
+            .run((rawRumor, peerId2))
+            .getOrElseF(
+              IO.raiseError(new Exception("failed to handle rumor"))
+            )
```

Comment:
```suggestion
          _ <- handler.run((rawRumor, peerId2)).value
```

Reply:
@cngo-github Perhaps a refactor closer to what you had is
```
          _ <- handler
            .run((rawRumor, peerId2))
            .getOrRaise(new Exception("failed to handle rumor"))
```
if `None` is considered an error condition.

Reply:
Changed to

```
          _ <- handler
            .run((rawRumor, peerId2))
            .getOrRaise(new Exception("failed to handle rumor"))
```